doc/SAI-Proposal-ACL-1.md

```diff
@@ -0,0 +1,89 @@
+SAI access control lists (ACL) enhancements for SAI 1.0 release
+-------------------------------------------------------------------------------
+
+ Title       | SAI ACL Model - Enhancements
+-------------|----------------------
+ Authors     | Cavium Inc.
+ Status      | In review
+ Type        | Standards track
+ Created     | 08/09/2016
+ SAI-Version | 1.0.0
+
+-------------------------------------------------------------------------------
+
+## Overview ##
+SAI Access Control Lists (ACL) object implements ACL management functions. In SAI 0.9.1 through 0.9.5 versions, SAI ACL contained three types of objects, ACL table, ACL entry and ACL counter. The ACL table contains a number of ACL entries. Each ACL table defines a set of unique matching fields for all its ACL entries. A packet can match rules in different ACL tables and take non-conflicting actions from all the matched rules. However, within an ACL table, if a packet matches multiple rules, only the actions from the rule of highest priority are executed. ACL counters can also be created and attached to an ACL entry in order to counter the number of packets or bytes that match the ACL entry. The initial version of ACL table object has several ambiguities and limitations. We propose the following enhancements to address some of those and propose a generic and simple ACL model for operators:
```

Comment:
Від: Shah, Zubin [mailto:Zubin.Shah@cavium.com] 
Надіслано: Wednesday, October 5, 2016 12:30 AM
Кому: Itai Baz <itaib@mellanox.com>
Копія: Shah, Zubin <Zubin.Shah@cavium.com>; Guohan Lu <gulv@microsoft.com>; Matty Kadosh <mattyk@mellanox.com>; Yakiv Huryk <yakivh@mellanox.com>; Jain, Atit <Atit.Jain@cavium.com>
Тема: Re: question re multi group binding
 
HI Itai,
Extremely sorry for the delay in responding. It seems this message got lost while i was on vacation. Please see my answers inline.
 
Thanks,
Zubin
 
On Sep 21, 2016, at 7:35 PM, Itai Baz <itaib@mellanox.com> wrote:
 
Hi Zubin,

I have a question re multi group binding to the same binding point (also posted it on the github).
With the old API, we had the definition :
* Within one ACL group, only one ACL entry will be hit which is based on the
* table priority as well as ACL entry priority within the table.
* Across the ACL groups, one hit ACL entry from each group will be selected
* and all non-conflicting actions from them will be performed.

Zubin>> Just to clarify that during the community review , it was clear that there were folks who had grave objections to support “resolution” of non-conflicting actions across ACL tables of a specific group, hence to simplify the behavior was explicitly discussed to be what you wrote below (highlighted in green). I know your questions relates to non-conflicting actions across ACL group(s).
 
With the new API, I can only see how to bind one group to each binding point, since each binding point has just one object ID, either a group or a table.
Within the group, tables will be evaluated according to table priority, and so there can only be one match per binding point.

So not sure if i'm missing something - how can we support multiple matches, one from each group, with the new API ?
Should we change the binding point to accept object list, thus allowing multiple groups to be bounded ?

 
Zubin>> I am completely okay and would be able to accommodate this change of accepting the attribute list at every binding point. However, I’d suggest keeping it optional, because there are folks who would disagree to this. What do you guys in the CC list think?